pkg/logger/context.go

```diff
@@ -0,0 +1,127 @@
+package logger
+
+import "time"
+
+// Context is data held about a certain logging event for a particular sink.
+// The data can be stored in any way that seems suitable for efficiently
+// composing a logging message for that sink.
+//
+// Most methods return the context itself. It is up to the implementation to
+// take advantage of this or use pointers to the same object throughout.
+//
+// It is up to the user of this type to honor the specification of feeding the
+// Context around to itself and calling the Append... methods on the return
+// value from any other such method.
+//
+// Good:
+// 	ctx.AppendString("hello", "world").WriteOut(logger.LevelDebug, "")
+//
+// Good:
+// 	ctx = ctx.AppendString("hello", "world")
+// 	ctx.WriteOut(logger.LevelDebug, "")
+//
+// Bad:
+// 	ctx.AppendString("hello", "world") // undefined behaviour
+// 	ctx.WriteOut(logger.LevelDebug, "")
+type Context interface {
+	// WriteOut sends the log message with the collected data from all the
+	// Append... methods
+	WriteOut(level Level, message string)
+
+	// SetScope sets the scope value for this context.
+	//
+	// Calling this method multiple times shall override the previous value.
+	// An empty string signifies to unset this field.
+	//
+	// In contrast to AppendString, the logging sink is allowed to render this
+	// differently. E.g. some may render it as yet another field named "scope",
+	// others may render it as a specific HTTP header in a request.
+	SetScope(value string) Context
+	// SetCaller adds a caller and its line value to this context.
+	//
+	// Calling this method multiple times shall override the previous value.
+	// An empty string on the file name signifies to unset this field.
+	//
+	// In contrast to AppendString, the logging sink is allowed to render this
+	// differently. E.g. some may render it as yet another fields named "caller"
+	// and "line", others may render it as a specific HTTP header in a request.
+	SetCaller(file string, line int) Context
+	// SetError adds an error value to this context.
```

Comment:
Same as the one before for SetScope
```suggestion
	// SetError sets the error value for this context.
```